sourcegraph/sourcegraph.proto

```diff
@@ -288,58 +289,115 @@ message RepoList {
 
 service RepoBadges {
 	// ListBadges lists the available badges for repo.
-	rpc ListBadges(RepoSpec) returns (BadgeList);
+	rpc ListBadges(RepoSpec) returns (BadgeList) {
+		option (google.api.http) = {
+			get: "/v1/repo_badges/list_badges"
```

Comment:
My rational behind this was that Twitter, Tumblr, and various other services all do it this way. I think the intent of placing it inside the URL is that removal of the `v1` API completely would have users hitting 404 not found, rather than accidentally running into a new API URL under the same name.

What use does the version keep if it's _optional_? Do we plan to keep full backwards compatibility for the API forever?